File: branches/Medusa/Source/Fantasy.Jobs/JobLogFileService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Fantasy.ServiceModel;
using System.IO;
using Fantasy.IO;

namespace Fantasy.Jobs
{
    public class JobLogFileService : LogFileService
    {


        public override void InitializeService()
        {
            IJobEngine engine = (IJobEngine)((IServiceProvider)this.Site).GetService(typeof(IJobEngine));

            string filePath = string.Format("{0}\\{1}.xlog", engine.JobDirectory, engine.JobId);
            FileStream fs = LongPathFile.Open(filePath, FileMode.Append, FileAccess.Write, FileShare.Read);
            _writer = new StreamWriter(fs, Encoding.UTF8);
            WriteStart();

            base.InitializeService();
        }

        public override void UninitializeService()
        {
            base.UninitializeService();
            this._writer.Close();
        }

        public static string GetLogFilePath(string jobsDirectory, Guid jobId)
        {
            return string.Format("{0}\\{1}\\{1}.xlog", jobsDirectory, jobId);
        }

        private StreamWriter _writer;

        protected override StreamWriter GetWriter()
        {
            return this._writer;
        }
    }
}

File: AutoConverted/Fantasy.Jobs/JobLogFileService.java
package Fantasy.Jobs;

import Fantasy.ServiceModel.*;
import Fantasy.IO.*;

public class JobLogFileService extends LogFileService
{


	@Override
	public void InitializeService()
	{
		IJobEngine engine = (IJobEngine)((IServiceProvider)this.Site).GetService(IJobEngine.class);

		String filePath = String.format("%1$s\\%2$s.xlog", engine.getJobDirectory(), engine.getJobId());
		FileStream fs = LongPathFile.Open(filePath, FileMode.Append, FileAccess.Write, FileShare.Read);
		_writer = new StreamWriter(fs, Encoding.UTF8);
		WriteStart();

		super.InitializeService();
	}

	@Override
	public void UninitializeService()
	{
		super.UninitializeService();
		this._writer.Close();
	}

	public static String GetLogFilePath(String jobsDirectory, Guid jobId)
	{
		return String.format("%1$s\\%2$s\\%2$s.xlog", jobsDirectory, jobId);
	}

	private StreamWriter _writer;

	@Override
	protected StreamWriter GetWriter()
	{
		return this._writer;
	}
}